openghg/store/base/_datasource.py

```diff
@@ -567,53 +628,9 @@ def load_dataset(bucket: str, key: str) -> xr.Dataset:
         Returns:
             xarray.Dataset: Dataset from NetCDF file
         """
-        from openghg.objectstore import get_object_data_path
-
-        file_path = get_object_data_path(bucket, key)
-        # This works
-        # TODO - this should be changed to use open_dataset really
-        # but open_dataset causes issues due to internal xarray caching when
-        # we try and write out to the same file
-        return xr.load_dataset(file_path)
-        # This does not
-        # return xr.open_dataset(file_path)
-
-    @classmethod
-    def from_data(cls: Type[T], bucket: str, data: Dict, shallow: bool) -> T:
-        """Construct a Datasource from JSON
-
-        Args:
-            bucket: Bucket containing data
-            data: JSON data
-            shallow: Load only the JSON data, do not retrieve data from the object store
-        Returns:
-            Datasource: Datasource created from JSON
-        """
-        from openghg.util import timestamp_tzaware
-
-        d = cls()
-        # TODO: May want to merge these steps within the @classmethod
-        # into __init__() so this is not added twice.
-        d._uuid = data["UUID"]
-        d._creation_datetime = timestamp_tzaware(data["creation_datetime"])
-        d._metadata = data["metadata"]
-        d._stored = data["stored"]
-        d._data_keys = data["data_keys"]
-        d._data = {}
-        d._data_type = data["data_type"]
-        d._latest_version = data["latest_version"]
-        d._bucket = bucket
-
-        if d._stored and not shallow:
-            for date_key in d._data_keys["latest"]["keys"]:
-                data_key = d._data_keys["latest"]["keys"][date_key]
-                d._data[date_key] = Datasource.load_dataset(bucket=bucket, key=data_key)
-
-        d._stored = False
-
-        d.update_daterange()
-
-        return d
+        raise NotImplementedError(
+            "Loading of data directly from Datasource no longer supported. Use memory_store to access data stored in zarr store."
+        )
 
     def define_version_key(self, version: str) -> str:
```

Comment:
`define_version_key`, `define_data_key`, and `define_backup_version` aren't used anywhere